circle/functions

```diff
@@ -496,19 +496,32 @@ chart_update_version() {
 chart_package() {
   local chart=${1}
   local name=${chart##*/}
+  local subpath=${chart%%/*}
   local src=${CIRCLE_WORKING_DIRECTORY}/${chart}
-  local dest=${CHART_OUTPUT_DIR}/${chart%%/*}
+  local dest=${CHART_OUTPUT_DIR}/${subpath}
+  local repo_index=${dest}/index.yaml
 
   mkdir -p $dest
 
+  # download the repo index.yaml file if it does not exist
+  MERGE_INDEX=
+  if [[ -f $repo_index ]]; then
```

Comment:
when would the repo index already be downloaded in the circle ci context?

Reply:
the script `helm-package-charts.sh` loops over every chart in the repo and calls `helm-package-charts.sh`. So for the first chart it processes in a path (incubator/stable) it will download the `index.yaml`. Subsequent calls to `package_chart` do not need to download the index.yaml as it already exists.

Reply:
ah I see, thanks!